File: ReflexTest TAP/Graph/ScriptTemplate/Ceil/CGCeil.js
/**
 * @file CGCeil.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGCeil.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGCeil extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    return Math.ceil(this.inputs[0]());
  }
}

exports.CGCeil = CGCeil;

File: ReflexTest TAP/Graph/ScriptTemplate/Ceil/CGCeil.lua
local CGCeil= CGCeil or {}
CGCeil.__index = CGCeil

function CGCeil.new()
    local self = setmetatable({}, CGCeil)
    self.inputs = {}
    return self
end

function CGCeil:setInput(index, func)
    self.inputs[index] = func
end

function CGCeil:getOutput(index)
    return math.ceil(self.inputs[0]())
end

return CGCeil
